contracts/distribution/dao-rewards-distributor/src/contract.rs

```diff
@@ -594,6 +597,27 @@ fn execute_update_owner(
     Ok(Response::new().add_attributes(ownership.into_attributes()))
 }
 
+fn execute_unsafe_force_withdraw(
```

Comment:
regarding this force withdraw, given that the funds are in the contract already, couldn't we call `execute_fund` to re-account for them?

Reply:
yeah so unfortunately, because we don't store claimed and pending rewards anywhere, it gets quite complicated to piece back together the missed rewards. we store rewards in the 'per-unit-voting-power' units, meaning that we'd need to take into account all total voting power changes since the stream began in order to compute what the `total_earned_puvp` *should* be AND what it translates to in terms of absolute reward tokens. this is further made more complex by the fact that distributions may have transitioned between different emission rates. there's no automatic way to just "fix" the existing distributions.

to your point in the review comment, the distributor is now already in a detached state because the state does not correspond with the amount of rewards in the contract. the distributor thinks rewards have been distributed that have not been, leaving excess funds in the contract that should have been distributed but were not. so right now, `funded_amount` is correct, but `ends_at` is incorrect, since it was computed assuming `funded_amount` is fully distributed. however, a distribution will distribute less than `funded_amount`, with some left over.

i think the ideal way to remedy this is:
- pause all distributions, which resets the dates
- withdraw all undistributed funds from the distributions, which resets `funded_amount` to 0
- upgrade the contract with the decimal fix that *does* distribute partial-interval amounts correctly
- force withdraw any remaining funds that should have been distributed
- restart the distributions and fund them again

if you don't want to restart all of them, you could probably manually piece together every funding event, loop through everyone's pending rewards queries, and calculate the missed rewards... but i think that is prone to too much error, and it's better to start over.

Reply:
yeah, that's fair. the hard reset logic you describe with pause -> withdraw -> restart & fund is pretty much what I had in mind as well!